typescript/src/test/createServer.ts

```diff
@@ -1,34 +1,36 @@
 import createServer from "../server";
 import ServerRx from "../server/rxjs-server";
 import Server from "../server/callback-server";
-import createGame from"../game_loop";
+import createGame from "../game_loop";
 
-export function createTestServer(style: "rxjs", port: number): Promise<ServerRx>;
-export function createTestServer(style: "callback", port: number): Promise<Server>;
-export function createTestServer(style: "rxjs" | "callback", port: number): Promise<ServerRx | Server> {
-    return new Promise((resolve, reject) => {
-        const server = createServer(style, port);
-        createGame(style, server);
+export function createTestServer(
+  style: "rxjs",
+  port: number
+): Promise<ServerRx>;
+export function createTestServer(
+  style: "callback",
+  port: number
+): Promise<Server>;
+export async function createTestServer(
+  style: "rxjs" | "callback",
+  port: number
+): Promise<ServerRx | Server> {
+  const server = createServer(style, port);
+  createGame(style, server);
 
-        if (style === "callback") {
-            server.on("listening", (e: Error) => {
-                if (e) {
-                    reject(e);
-                } else {
-                    resolve(server);
-                }
-            });
+  if (style === "callback") {
+    return new Promise((resolve, reject) => {
+      server.on("listening", (e: Error) => {
+        if (e) {
+          reject(e);
         } else {
-            (server as ServerRx).listening.subscribe({
-                next: (res) => {
-                    if (res) {
-                        resolve(server);
-                    }
-                },
-                error: reject,
-            });
+          resolve(server);
         }
-    })
+      });
+    });
+  } else {
+    // RxJS server
+    await (server as ServerRx).listening;
```

Comment:
The Rx server was changed just to use a Promise. An observable didn't make sense here, really.